src/common/pico_util/include/pico/util/pheap.h

```diff
@@ -105,49 +135,154 @@ static pheap_node_id_t ph_merge_nodes(pheap_t *heap, pheap_node_id_t a, pheap_no
     }
 }
 
+/**
+ * Allocate a new node from the unused space in the heap
+ *
+ * \param heap the heap
+ * \return and identifier for the node, or 0 if the heap is full
+ */
 static inline pheap_node_id_t ph_new_node(pheap_t *heap) {
     if (!heap->free_head_id) return 0;
     pheap_node_id_t id = heap->free_head_id;
-    heap->free_head_id = ph_get_node(heap, id)->sibling;
+    pheap_node_t *hn = ph_get_node(heap, id);
+    heap->free_head_id = hn->sibling;
     if (!heap->free_head_id) heap->free_tail_id = 0;
+    hn->child = hn->sibling = hn->parent = 0;
     return id;
 }
 
-// note this will callback the comparator for the node
-// returns the (new) root of the heap
-static inline pheap_node_id_t ph_insert(pheap_t *heap, pheap_node_id_t id) {
+/**
+ * Inserts a node into the heap.
+ *
+ * This method inserts a node (previously allocated by ph_new_node())
+ * into the heap, determining the correct order by calling
+ * the heap's comparator
+ *
+ * \param heap the heap
+ * \param id the id of the node to insert
+ * \return the id of the new head of the pairing heap (i.e. node that compares first)
+ */
+static inline pheap_node_id_t ph_insert_node(pheap_t *heap, pheap_node_id_t id) {
     assert(id);
     pheap_node_t *hn = ph_get_node(heap, id);
     hn->child = hn->sibling = hn->parent = 0;
     heap->root_id = ph_merge_nodes(heap, heap->root_id, id);
     return heap->root_id;
 }
 
+/**
+ * Returns the head node in the heap, i.e. the node
+ * which compares first, but without removing it from the heap.
+ *
+ * \param heap the heap
+ * \return the
+ */
 static inline pheap_node_id_t ph_peek_head(pheap_t *heap) {
     return heap->root_id;
 }
 
-pheap_node_id_t ph_remove_head_reserve(pheap_t *heap, bool reserve);
+/**
+ * Remove the head node from the pairing heap. This head node is
+ * the node which compares first in the logical ordering provided
+ * by the comparator.
+ *
+ * Note that in the case of reserve = false, the returned id is no longer
```

Comment:
"in the case of reserve = false" - does this need to be rewritten now that the parameter is named `free` ?